Storage/src/SigningHelper.php

```diff
@@ -882,4 +896,46 @@ private function buildQueryString(array $input)
 
         return implode('&', $q);
     }
+
+    /**
+     * Retry logic for signBlob
+     *
+     * @param callable $signBlobFn  A callable that perform the actual signBlob operation.
+     * @param string $resourceName The resource name for logging or retry strategy determination.
+     * @param array $args Arguments for the operations, include preconditions
+     * @return string The signature genarated by signBlob.
+     * @throws ServiceException If non-retryable error occur.
+     * @throws \RuntimeException If retries are exhausted.
+     */
+    private function retrySignBlob(callable $signBlobFn, string $resourceName = 'signBlob', array $args = [])
+    {
+        $attempts = 0;
+        $maxRetries = 5;
+        $invocationId = Uuid::uuid4()->toString();
+        $args['retryStrategy'] = StorageClient::RETRY_ALWAYS;
+
+        // Generate a retry decider function using the RetryTrait logic.
+        $retryDecider = $this->getRestRetryFunction($resourceName, 'execute', $args);
+
+        while ($attempts < $maxRetries) {
+            $attempts++;
```

Comment:
You're right, I couldn't find a maxRetries or maxAttempts option in the documentation either. That's why I added the maxAttempt directly.

Reply:
I'll approve from my side; just need @bshaffer to weigh in here when he has a chance.

Reply:
If I'm reading this correctly, the current code does not have any check for `maxAttempts`, so it seems like this would result in an infinite loop. There is no test case for a retryable error meeting its max attempts. I feel like we definitely need a retry max attempts.